File: src/main/java/com/gangofconnectfour/powerfourservice/api/in/SessionDtoIn.java
package com.gangofconnectfour.powerfourservice.api.in;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.Email;

@Validated
@Data
@AllArgsConstructor
@NoArgsConstructor
public class SessionDtoIn {

    @Email
    private String email;
    private String encryptedPass;
    private String nickname;
    private String token;
}

File: src/main/kotlin/com/gangofconnectfour/powerfourservice/api/in/SessionDtoIn.kt
package com.gangofconnectfour.powerfourservice.api.`in`

import org.springframework.validation.annotation.Validated
import javax.validation.constraints.Email

@Validated
class SessionDtoIn {

    @Email
    var email : String = ""
    var encryptedPass : String = ""
    var nickname : String = ""
    var token : String = ""


    constructor()
    constructor(email: String, encryptedPass: String, nickname: String, token: String) {
        this.email = email
        this.encryptedPass = encryptedPass
        this.nickname = nickname
        this.token = token
    }


}